File: PGMS Backend/src/main/java/com/pgm/pgm_Backend/service/AppointmentService.java
package com.pgm.pgm_Backend.service;

import com.pgm.pgm_Backend.model.Appointment;
import java.util.List;

public interface AppointmentService {
    Appointment createAppointment(Appointment appointment);

    List<Appointment> getAllAppointments();

    List<Appointment> getAppointmentsByAdminId(Long adminId);

    Appointment getAppointmentById(Long id);

    Appointment updateAppointmentStatus(Long id, String status);

    void deleteAppointment(Long id);
}

File: PGMS/PGMS Frontend/src/services/appointmentService.js
import { apiGet, apiPost, apiPut } from './apiHelper'

const APPOINTMENTS_BASE_URL = '/appointments'

export const createAppointment = async (appointmentData) => {
    return await apiPost(APPOINTMENTS_BASE_URL, appointmentData)
}

export const getAllAppointments = async () => {
    return await apiGet(APPOINTMENTS_BASE_URL)
}

export const getAppointmentById = async (id) => {
    return await apiGet(`${APPOINTMENTS_BASE_URL}/${id}`)
}

export const updateAppointmentStatus = async (id, status) => {
    return await apiPut(`${APPOINTMENTS_BASE_URL}/${id}/status`, { status })
}
